subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -1012,11 +1017,16 @@ void ztest_run_all(const void *state) { ztest_api.run_all(state); }
 
 void __weak test_main(void)
 {
+#ifdef CONFIG_ZTEST_STANDALONE_EXECUTION
+	z_init_mock();
+#endif
+
 	ztest_run_all(NULL);
 
 	ztest_verify_all_test_suites_ran();
 }
 
+#ifndef CONFIG_ZTEST_STANDALONE_EXECUTION
 #ifndef KERNEL
 int main(void)
```

Comment:
I think it may be a bit cleaner if we rename the current `main` functions to `ztest_main`  (or some better name) and create new `main` functions that call the renamed `ztest_main()`.  So in your standalone app, you can invoke `ztest_main` which does everything that the current `main` does.  This way we don't need to sprinkle around the inits and report functions throughout the file.

So something like
```
int ztest_main(void) 
{
	z_init_mock();
	test_main();
	end_report();
	return test_status;
};

#ifndef CONFIG_ZTEST_STANDALONE_EXECUTION
int main(void) 
{ 
  return ztest_main(); 
};
#endif 
```

Would this accomplish the behavior you're looking for?

Reply:
+1

Reply:
Good idea, I'll do this

Reply:
I just pushed new commits
Let me know what you think